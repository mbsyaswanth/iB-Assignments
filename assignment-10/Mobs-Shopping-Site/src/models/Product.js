class Product {
  constructor(productObj) {
    this.availableSizes = productObj.availableSizes;
    this.currencyFormat = productObj.currencyFormat;
    this.currencyId = productObj.currencyId;
    this.description = productObj.description;
    this.id = productObj.id;
    this.installments = productObj.installments;
    this.isFreeShipping = productObj.isFreeShipping;
    this.price = productObj.price;
    this.sku = productObj.sku;
    this.style = productObj.style;
    this.title = productObj.title;
    this.image = productObj.image;
  }
}

export default Product;
