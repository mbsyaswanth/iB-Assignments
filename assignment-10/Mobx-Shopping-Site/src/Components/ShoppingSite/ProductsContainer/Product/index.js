import React, { Component } from "react";
import { observer } from "mobx-react";
import { observable } from "mobx";

import {
  ProductContainer,
  FreeShipping,
  ProductInfo,
  ProductTitle,
  Line,
  ProductPrice,
  ProductInstallments,
  ProductCartBtn,
  Select,
  Option
} from "./styledComponents";

@observer
class Product extends Component {
  @observable size = this.props.product.availableSizes[0];
  onSizeChange = event => {
    console.log(event.target.value);
    this.size = event.target.value;
  };

  handleClick = () => {
    this.props.store.addToCart(this.props.product.id, this.size);
  };

  render() {
    const {
      isFreeShipping,
      image,
      title,
      price,
      installments,
      availableSizes
    } = this.props.product;

    return (
      <ProductContainer>
        <FreeShipping id="freeshipping" isFreeShipping={isFreeShipping}>
          Free shipping
        </FreeShipping>
        <div>
          <img alt="productimage" src={image} />
        </div>
        <ProductInfo>
          <ProductTitle id="title1">{title}</ProductTitle>
          <Line />
          <ProductPrice id="price">$ {price}</ProductPrice>
          <ProductInstallments id="installments">
            or {installments}x ${(price / installments).toFixed(2)}
          </ProductInstallments>
          <div>
            <Select value={this.size} onChange={this.onSizeChange}>
              {availableSizes.map(size => {
                return (
                  <Option value={size} key={size}>
                    {size}
                  </Option>
                );
              })}
            </Select>
          </div>
        </ProductInfo>
        <ProductCartBtn id="addtocart" onClick={this.handleClick}>
          Add to cart
        </ProductCartBtn>
      </ProductContainer>
    );
  }
}

export default Product;
