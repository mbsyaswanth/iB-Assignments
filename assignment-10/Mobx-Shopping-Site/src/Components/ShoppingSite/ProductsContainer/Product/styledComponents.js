import styled from "styled-components";

export const ProductContainer = styled.div`
  position: relative;
  padding: 10px;
  margin: 5px;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 333px;
  background-color: white;
  transition: 0.2s;
  :hover {
    box-shadow: 0px 0px 2px 1px grey;
  }
`;

export const FreeShipping = styled.div`
  display: ${props => (props.isFreeShipping ? "block" : "none")};
  position: absolute;
  top: 10px;
  right: 10px;
  color: white;
  padding: 2px;
  background-color: black;
`;

export const ProductInfo = styled.div`
  display: flex;
  flex-direction: column;
  height: 160px;
  justify-content: space-evenly;
  align-items: center;
  margin-top: 10px;
`;

export const ProductTitle = styled.div`
  font-size: 1.2rem;
  height: 2.4rem;
`;

export const Line = styled.div`
  width: 30px;
  height: 2px;
  background-color: goldenrod;
`;

export const ProductPrice = styled.div`
  font-weight: 600;
  font-size: 1.6rem;
`;

export const ProductInstallments = styled.div`
  color: grey;
  font-weight: 600;
`;

export const ProductCartBtn = styled.button`
  width: 333px;
  padding: 10px;
  font-size: 1.1rem;
  color: white;
  background-color: black;
  font-weight: 500;
  border-style: none;
  cursor: pointer;
  transition: 0.2s;
  :focus {
    outline: none;
  }
  :active {
    border-style: none;
  }
  ${ProductContainer}:hover & {
    background-color: goldenrod;
  }
`;

export const Select = styled.select``;

export const Option = styled.option``;
