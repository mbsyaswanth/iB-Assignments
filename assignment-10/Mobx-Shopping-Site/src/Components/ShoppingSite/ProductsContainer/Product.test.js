import React from "react";
import { shallow, mount, render } from "enzyme";
import Product from "./Product";

const product = {
  availableSizes: ["S", "XS"],
  currencyFormat: "$",
  currencyId: "USD",
  description: "4 MSL",
  id: 12,
  installments: 9,
  isFreeShipping: true,
  price: 10.9,
  sku: 12064273040195392,
  style: "Black with custom print",
  title: "Cat Tee Black T-Shirt",
  image:
    "https://react-shopping-cart-67954.firebaseapp.com/static/media/12064273040195392_1.2995d79a.jpg"
};

describe("Product", () => {
  it("should load the given product according to the props.", () => {
    const component = shallow(<Product product={product} />);
    console.log(component.find("#title").text());
    expect(component.find("#title").text()).toBe("Cat Tee Black T-Shirt");
    component.unmount();
  });
});
