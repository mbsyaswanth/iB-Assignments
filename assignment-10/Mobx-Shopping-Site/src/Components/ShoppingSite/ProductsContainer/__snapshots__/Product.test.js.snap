// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingSite should render correctly in "debug" mode 1`] = `
<styled.div>
  <styled.div
    isFreeShipping={true}
  >
    Free shipping
  </styled.div>
  <div>
    <img
      alt="productimage"
      src="https://react-shopping-cart-67954.firebaseapp.com/static/media/12064273040195392_1.2995d79a.jpg"
    />
  </div>
  <styled.div>
    <styled.div>
      Cat Tee Black T-Shirt
    </styled.div>
    <styled.div />
    <styled.div>
      $ 
      10.9
    </styled.div>
    <styled.div>
      or 
      9
      x $
      1.21
    </styled.div>
    <div>
      <styled.select
        onChange={[Function]}
        value="S"
      >
        <styled.option
          key="S"
          value="S"
        >
          S
        </styled.option>
        <styled.option
          key="XS"
          value="XS"
        >
          XS
        </styled.option>
      </styled.select>
    </div>
  </styled.div>
  <styled.button
    onClick={[Function]}
  >
    Add to cart
  </styled.button>
</styled.div>
`;
